home-service/src/main/java/com/softserveinc/ita/homeproject/homeservice/service/impl/VoteServiceImpl.java

```diff
@@ -0,0 +1,190 @@
+            } else {
+                questionVoteDtos.add(mapper.convert(qv, ReadMultipleChoiceQuestionVoteDto.class));
+            }
+        }
+        ReadVoteDto readVoteDto = new ReadVoteDto();
+        readVoteDto.setId(reSavedVote.getId());
+        readVoteDto.setQuestionVoteDtos(questionVoteDtos);
+        return readVoteDto;
+    }
+
+    private void validatePollStatus(Long pollId) {
+        PollStatus pollStatus = pollRepository.findById(pollId).orElseThrow(() -> new NotFoundHomeException(
+            String.format(NOT_FOUND_MESSAGE, "Poll", pollId))).getStatus();
+        if (!pollStatus.equals(PollStatus.ACTIVE)) {
+            throw new BadRequestHomeException(
+                String.format(POLL_STATUS_VALIDATION_MESSAGE, pollStatus));
+        }
+    }
+
+    private void validateReVoting(Long pollId, User currentUser) {
+        if (voteRepository.findByPollIdAndUser(pollId, currentUser) != null) {
+            throw new BadRequestHomeException(
+                String.format(TRYING_TO_REVOTE_MESSAGE, pollId));
+        }
+    }
+
+    private void validateQuestionVotesCount(Long pollId, CreateVoteDto voteDto) {
+        int questionVotesCount = voteDto.getQuestionVoteDtos().size();
+        int questionsCount = pollRepository.findById(pollId).orElseThrow(() -> new NotFoundHomeException(
+            String.format(NOT_FOUND_MESSAGE, "Poll", pollId))).getPollQuestions().size();
+        if (questionVotesCount != questionsCount) {
+            throw new BadRequestHomeException(
+                String.format(WRONG_QUESTIONS_COUNT_FOR_POLL_MESSAGE, pollId));
+        }
+    }
+
+    private void validatePollQuestionsMatching(Long pollId, CreateVoteDto voteDto) {
+        List<CreateQuestionVoteDto> questionVotes = voteDto.getQuestionVoteDtos();
+        int questionsCount = questionVotes.size();
+        int controlNumber = 0;
+        for (CreateQuestionVoteDto questionVote : questionVotes) {
+            Long questionId = questionVote.getQuestionId();
+            PollQuestion question = questionRepository.findById(questionId).orElseThrow(
+                () -> new NotFoundHomeException(String.format(NOT_FOUND_MESSAGE, "Poll question", questionId)));
+            Long questionPollId = question.getPoll().getId();
+            if (questionPollId.equals(pollId)) {
+                controlNumber++;
+            }
+        }
+        if (questionsCount != controlNumber) {
+            throw new BadRequestHomeException(
+                String.format(WRONG_QUESTIONS_FOR_POLL_MESSAGE, pollId));
+        }
+    }
+
+    private void validateAnswerCounts(CreateVoteDto voteDto) {
+        List<CreateQuestionVoteDto> questionVotes = voteDto.getQuestionVoteDtos();
+        for (CreateQuestionVoteDto questionVote : questionVotes) {
+            Long questionId = questionVote.getQuestionId();
+            PollQuestion question = questionRepository.findById(questionId).orElseThrow(
```

Comment:
How you know enabled this question or not?

Reply:
Check for enabled is done